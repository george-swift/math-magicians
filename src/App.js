import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>Calculator With React</p>
      </header>
    </div>
  );
}

export default App;
