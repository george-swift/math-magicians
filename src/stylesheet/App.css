body {
  background-color: #e1e0df;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

.container {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.calculator {
  width: 500px;
  height: 600px;
  background-color: black;
}

.group-one,
.group-two,
.group-three,
.group-four,
.group-five {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 3px;
  height: 19.15%;
  margin-bottom: 2.45px;
}

button {
  border-color: transparent;
  font-size: 1.4em;
}

.clear:active,
.number:active,
.decimal:active {
  background-color: #b2b2b2;
}

#number-0 { grid-column: 1/3; }

.operator { background-color: #ff8f1f; }

.operator:active { transform: translateY(1px); }

@media screen and (max-width: 375px) {
  .calculator {
    width: 400px;
    height: 500px;
  }
}

.problem,
.result {
  height: 50%;
  padding: 0.75rem 0.4rem 0.4rem;
  font-weight: 300;
}

.problem {
  font-size: 2em;
}

.result {
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
  font-size: 3em;
}
