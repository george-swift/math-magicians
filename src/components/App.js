import '../stylesheet/App.css';
import { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

export default function App() {
  const calculatorData = {
    total: null,
    next: null,
    operation: null,
  };

  const [data, setData] = useState(calculatorData);

  const handleClick = (buttonName) => setData((data) => calculate(data, buttonName));

  const { total, next, operation } = data;

  return (
    <div className="container">
      <div className="row flex-column calculator">
        <Display operation={operation} result={next || total || '0'} />
        <ButtonPanel clickHandler={handleClick} />
      </div>
    </div>
  );
}
