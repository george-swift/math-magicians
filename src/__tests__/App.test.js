import renderer from 'react-test-renderer';
import App from '../components/App';

describe('App component', () => {
  test('Should render child components and routes correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
