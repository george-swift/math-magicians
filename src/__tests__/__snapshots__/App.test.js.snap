// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should render child components and routes correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light bg-gradient sticky-top"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        <span
          className="me-2"
        >
          Math Magicians
        </span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 96l16-32 32-16-32-16-16-32-16 32-32 16 32 16 16 32zM80 160l26.66-53.33L160 80l-53.34-26.67L80 0 53.34 53.33 0 80l53.34 26.67L80 160zm352 128l-26.66 53.33L352 368l53.34 26.67L432 448l26.66-53.33L512 368l-53.34-26.67L432 288zm70.62-193.77L417.77 9.38C411.53 3.12 403.34 0 395.15 0c-8.19 0-16.38 3.12-22.63 9.38L9.38 372.52c-12.5 12.5-12.5 32.76 0 45.25l84.85 84.85c6.25 6.25 14.44 9.37 22.62 9.37 8.19 0 16.38-3.12 22.63-9.37l363.14-363.15c12.5-12.48 12.5-32.75 0-45.24zM359.45 203.46l-50.91-50.91 86.6-86.6 50.91 50.91-86.6 86.6z"
          />
        </svg>
      </a>
      <button
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNav"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <div
          className="navbar-nav ms-auto"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 9l-3-3V2h-2v2L8 1 0 9h2l1 5c0 .55.45 1 1 1h8c.55 0 1-.45 1-1l1-5h2zm-4 5H9v-4H7v4H4L2.81 7.69 8 2.5l5.19 5.19L12 14z"
                fillRule="evenodd"
              />
            </svg>
            <span
              className="ms-2"
            >
              Home
            </span>
          </a>
          <a
            className="nav-link"
            href="/calculator"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M368 48H144c-26.6 0-48 21.6-48 48v320c0 26.4 21.4 48 48 48h224c26.4 0 48-21.6 48-48V96c0-26.4-21.4-48-48-48zM200 416h-48v-48h48v48zm0-88h-48v-48h48v48zm0-88h-48v-48h48v48zm80 176h-48v-48h48v48zm0-88h-48v-48h48v48zm0-88h-48v-48h48v48zm80 176h-48V280h48v136zm0-176h-48v-48h48v48zm0-96H152V96h208v48z"
              />
            </svg>
            <span
              className="ms-2"
            >
              Calculator
            </span>
          </a>
          <a
            className="nav-link"
            href="/quote"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0H24V24H0z"
                  fill="none"
                />
                <path
                  d="M21 3c.552 0 1 .448 1 1v14c0 .552-.448 1-1 1H6.455L2 22.5V4c0-.552.448-1 1-1h18zM10.962 8.1l-.447-.688C8.728 8.187 7.5 9.755 7.5 11.505c0 .995.277 1.609.792 2.156.324.344.837.589 1.374.589.966 0 1.75-.784 1.75-1.75 0-.92-.711-1.661-1.614-1.745-.16-.015-.324-.012-.479.01v-.092c.006-.422.092-1.633 1.454-2.466l.185-.107-.447-.688zm4.553-.688c-1.787.775-3.015 2.343-3.015 4.093 0 .995.277 1.609.792 2.156.324.344.837.589 1.374.589.966 0 1.75-.784 1.75-1.75 0-.92-.711-1.661-1.614-1.745-.16-.015-.324-.012-.479.01 0-.313-.029-1.762 1.639-2.665z"
                />
              </g>
            </svg>
            <span
              className="ms-2"
            >
              Quote
            </span>
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container flex-column text-center pt-5"
  >
    <h1>
      Welcome to our page
    </h1>
    <div
      className="mt-4 card bg-dark text-white"
    >
      <img
        alt="A blackboard"
        className="w-100"
        src="roman-mager.jpg"
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          Math Magicians is a single page web app that provides an easy way of solving arithmetic problems. Built with React, React Router v6, Javascript, Bootstrap, HTML5/CSS3.
        </p>
        <a
          className="btn btn-warning"
          href="/calculator"
          onClick={[Function]}
        >
          Get Started
        </a>
      </div>
    </div>
  </div>,
]
`;
